engine/src/main/java/org/terasology/rendering/dag/RenderPipelineOptimizer.java

```diff
@@ -15,48 +15,67 @@
  */
 package org.terasology.rendering.dag;
 
-import com.google.common.collect.Lists;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
 import java.util.List;
+import java.util.Map;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  *
  */
-public class RenderPipelineOptimizer {
-    private RenderPipeline renderPipeline;
+class RenderPipelineOptimizer {
```

Comment:
Given that the pipeline to optimize is now out of sight I'd recommend to rename this class RenderTaskListGenerator and rename the method "optimize" to "generateTaskListFrom(orderedNodes)"